### DISCUSSION TITLE
Bandwidth Access

### DISCUSSION THREAD

**Discussion by noor-codes** [linuxdeepin/developer-center#2937] [Features Request & Ideas | 特性请求 & 头脑风暴]
## Bandwidth Access
It would be awesome to have a feature where we can whitelist and blacklist all the apps we choose to be able to access the internet or not.

Currently, there are apps on my pc that are bandwidth-heavy and I have no idea how to restrict that access.
Thanks.

**Comment by meiyixiang** [comment]
I think you're talking a bit like a firewall feature. I don't know if I understand it correctly

**Reply by noor-codes** [reply]
I think you are right, I found a solution for the moment.
https://github.com/safing/portmaster

**Reply by noor-codes** [reply]
The idea was simple, Just like we are on a limited data plan and don't have a lot of data to spare, iPhone has a feature where you can block the internet access going into specific apps which helps you organize where the data should be used. 

Thanks for the reply @meiyixiang

**Comment by meiyixiang** [comment]
In fact, we have a software in the commercial version that has the ability you mentioned, but it cannot be released in the community due to open source compliance issues. We will see if we can integrate the software in the community or put it on the app store later

**Reply by ghost** [reply]
> In fact, we have a software in the commercial version that has the ability you mentioned, but it cannot be released in the community due to open source compliance issues. We will see if we can integrate the software in the community or put it on the app store later

Can we have in the App Store some other apps like AI Voice? 🤔 that is mentioned as a speaking assistant in some ads and I understand is Union tech commercial, it even has keyboard shortcut (super+q if I remember right)

**Reply by noor-codes** [reply]
> In fact, we have a software in the commercial version that has the ability you mentioned, but it cannot be released in the community due to open source compliance issues. We will see if we can integrate the software in the community or put it on the app store later

That would be awesome, Please notify us if released

**Comment by meiyixiang** [comment]
ok
